web_src/js/components/DiffFileTree.vue

```diff
@@ -103,15 +104,23 @@ export default {
 
     this.hashChangeListener = () => {
       this.selectedFile = window.location.hash;
+      this.expandSelectedFile();
     };
     this.hashListener = window.addEventListener('hashchange', this.hashChangeListener);
     this.selectedFile = window.location.hash;
+    this.expandSelectedFile();
   },
   unmounted() {
     document.querySelector('.diff-toggle-file-tree-button').removeEventListener('click', this.toggleVisibility);
     window.removeEventListener('hashchange', this.hashChangeListener);
   },
   methods: {
+    expandSelectedFile() {
+      // expand file if the selected file is folded
+      const box = document.querySelector(this.selectedFile);
+      const folded = box.getAttribute('data-folded');
```

Comment:
`getAttribute` does not return booleans, so I think you may need to add ` === 'true'`.